nestkernel/conn_builder.cpp

```diff
@@ -341,6 +341,41 @@ nest::ConnBuilder::connect()
       "This connection rule does not support symmetric connections." );
   }
 
+  if ( kernel().model_manager.connector_requires_clopath_archiving(
+         synapse_model_id_ ) )
+  {
+    // get thread id
+    const thread tid = kernel().vp_manager.get_thread_id();
+    for ( GIDCollection::const_iterator tgid = targets_->begin();
```

Comment:
this loop seems to have complexity `O(n)` where `n` is total number of targets, and is hence not scalable to large networks. is it possible to delay this check until just before the actual creation of the connection, e.g., https://github.com/jstapmanns/nest-simulator/blob/clopath_stdp_synapse/nestkernel/connection_manager.cpp#L617, where connections have already been filtered by thread locality?

Reply:
Great suggestion. Implemented this with ddecf36